torch/multiprocessing/spawn.py

```diff
@@ -176,20 +181,28 @@ def start_processes(
     fn, args=(), nprocs=1, join=True, daemon=False, start_method="spawn"
 ):
     mp = multiprocessing.get_context(start_method)
-    error_queues = []
+    error_files = []
     processes = []
     for i in range(nprocs):
-        error_queue = mp.SimpleQueue()
+        # Each process is assigned a file to write tracebacks to.  We
+        # use the file being non-empty to indicate an exception
+        # occurred (vs an expected shutdown).  Note: this previously
+        # used a multiprocessing.Queue but that can be prone to
+        # deadlocks, so we went with a simpler solution for a one-shot
+        # message between processes.
+        tf = tempfile.NamedTemporaryFile(
+            prefix="pytorch-errorfile-", suffix=".pickle", delete=False
```

Comment:
how would you clean the files afterwards?

Reply:
They will be cleaned by whatever keeps tmp files clean on the machine since `tempfile` will pick a well-known location (either `/tmp` or following `$TMPDIR`).  We could clean them up but it also is perhaps a little useful having them around for later debugging.  Adding a delete when we consume them is easy enough.  I could go either way.  What would you prefer?